app/presenters/tree_builder_alert_profile_assign.rb

```diff
@@ -0,0 +1,29 @@
+class TreeBuilderAlertProfileAssign < TreeBuilder
+  def initialize(name, sandbox, build, **params)
+    @selected_nodes = params[:selected_nodes]
+    # need to remove tree info
+    TreeState.new(sandbox).remove_tree(name)
+    super(name, sandbox, build)
+  end
+
+  def x_get_tree_roots(count_only, _options)
+    roots = ExtManagementSystem.assignable.each_with_object({}) do |ems, nodes|
+      subtree = ems.children.flat_map(&:folders).each_with_object({}) do |folder, obj|
```

Comment:
a little concerned about `folders` here - but I think that is out of scope for this PR.

do a quick check but think keep this.

Reply:
Originally we were calling `folders_only + datacenters_only` on this level, and they look like:
```
  def folders_only
    folders.select { |f| !f.kind_of?(Datacenter) }
  end

  def datacenters_only
    folders.select { |f| f.kind_of?(Datacenter) }
  end
```
So I guess it's okay.

Reply:
yup. u.g.l.y.